packages/astro/src/integration/create-integration.ts

```diff
@@ -0,0 +1,112 @@
+import type { AstroIntegration } from 'astro';
+
+import { name as packageName } from '../../package.json';
+import type { AstroClerkIntegrationParams } from '../types';
+
+const buildEnvVarFromOption = (valueToBeStored: unknown, envName: string) => {
+  return valueToBeStored ? { [`import.meta.env.${envName}`]: JSON.stringify(valueToBeStored) } : {};
+};
+
+type HotloadAstroClerkIntegrationParams = AstroClerkIntegrationParams & {
+  clerkJSUrl?: string;
+  clerkJSVariant?: 'headless' | '';
+  clerkJSVersion?: string;
+};
+
+function createIntegration<P extends { mode: 'hotload' | 'bundled' }>({ mode }: P) {
+  return (
+    params?: P['mode'] extends 'hotload' ? HotloadAstroClerkIntegrationParams : AstroClerkIntegrationParams,
+  ): AstroIntegration => {
+    const { proxyUrl, isSatellite, domain, signInUrl, signUpUrl } = params || {};
+
+    // These are not provided when the "bundled" integration is used
+    const clerkJSUrl = (params as any)?.clerkJSUrl as string | undefined;
+    const clerkJSVariant = (params as any)?.clerkJSVariant as string | undefined;
+    const clerkJSVersion = (params as any)?.clerkJSVersion as string | undefined;
+
+    return {
+      name: '@clerk/astro/integration',
+      hooks: {
+        'astro:config:setup': ({ config, injectScript, updateConfig, logger, command }) => {
+          if (config.output === 'static') {
+            logger.error(`${packageName} requires SSR to be turned on. Please update output to "server"`);
```

Comment:
Why is SSR required? Can't it also work with CSR?

Reply:
Currently we support only SSR, because we are injecting the clerk-js script from the middleware. But i will add a ticket in order to explore this.